### DISCUSSION TITLE
Solis backup mode

### DISCUSSION THREAD

**Discussion by mart-mac** [wills106/homeassistant-solax-modbus#354] [Ideas]
There is an option of the Solis inverter to reserve a set % of the battery for a backup.  I think this would be a useful option if using the new Octopus Flux tariff, but would need to turn it on and off during the day

I think the registers for the Solis Backup Mode SOC 10-100% is 43024

but I can't find it for the Solis backup mode on/off

Is this something easy to be added?

**Comment by alienatedsec** [comment]
@mart-mac Where is that option on the inverter? Can you share a picture?

**Reply by mart-mac** [reply]
Can't get to the inverter at the moment for a pic, but its in the work modes.  You only see the percentage when it's turned on.  On the app you can turn it on and off, but not change the %

**Reply by alienatedsec** [reply]
@mart-mac i will need to read those registers and have a go first. @wills106 could then rely on the information or the commit (PR) after. 

I'll try tomorrow

**Reply by mart-mac** [reply]
Just took a picture, these are the 4 option included in the menu. The off grid option can already be controlled by the "energy storage control switch" entity
![20230317_083622](https://user-images.githubusercontent.com/106774411/225861713-8df6d5cb-04c8-4fdd-8296-5739f76ed293.jpg)

**Reply by alienatedsec** [reply]
So the Storage Mode will be one of those functions below. If there are any additional, those can be added too.

https://github.com/wills106/homeassistant-solax-modbus/blob/48ee7cee11b7bf06fb8a98529e618751bf68fc44/custom_components/solax_modbus/plugin_solis.py#L655-L664

@mart-mac Do you know how to read the register `43110` when you set it to `Backup` on your inverter?

Mine is on `35: "Timed Charge/Discharge"` at the moment

```
DEBUG:pysolarmanv5.pysolarmanv5:RECD: a5 15 00 10 15 00 2d 32 c3 fc f3 02 01 5b 43 69 00 ab 04 00 00 71 ee aa 63 01 03 02 00 23 f9 9d 2f 15
[35]
```

**Reply by mart-mac** [reply]
I have no idea what I'm doing really, lol.  This is my first time really using HA and anything modbus related.  

What would I need install on ha to do it, I'd like the learn

But I program cnc machine all day and I've been around computers since the early 90s so not total dumb to it, trying to learn

**Reply by alienatedsec** [reply]
I am also on self-use which is linked to `33` and `35` in the register `43110`. What is the `Backup` all about? Please explain in detail if possible, @mart-mac ?

**Reply by mart-mac** [reply]
Backup is designed to limit to battery usage down to a set percentage for use on the eps output in case of power cuts.  Its the only thing I can think of that can be used to reserve some of the battery for the even peak rate, while still allow solar charging of the batteries.  My idea it to turn it on in the morning after charging the batteries over night set at 50%, as long as the battery stays above 50% everything works the same.  If it drop to 50% I'll start pulling from the grid.  If solar increases and and charges the battery it'll revert back to normal.

**Reply by alienatedsec** [reply]
If you have a power cut, the battery and PV are useless as the inverter will cut your house from feeding the power. 

Unless an EPS (mentioned by you) is an additional feed independent of mains in your house and can be used to provide backup power whenever there is a power cut.

Do you have any documentation I can refer to?

**Reply by mart-mac** [reply]
The eps is a separate output on the bottom of the inverter and when active is live all the time, and will stay live during a power cut.  I have it connected to a double socket next to my inverter.

But my idea of using the backup mode and SOC is only to reserve the battery for the evening peak rate for the main output and not for the eps output.

I've just had a thought, this mode might not be visible without the eps enabled, ill check that out

**Reply by mart-mac** [reply]
I've just checked, it is visible even with the eps turned off

**Reply by alienatedsec** [reply]
@mart-mac can you post the Backup Mode SOC screen picture? mine is set to 80 when I read the `43024`

```
DEBUG:pysolarmanv5.pysolarmanv5:RECD: a5 15 00 10 15 00 52 32 c3 fc f3 02 01 24 ca 69 00 e1 0c 00 00 81 ee aa 63 01 03 02 00 50 b8 78 b9 15
[80]
```

**Reply by mart-mac** [reply]
![20230317_191223](https://user-images.githubusercontent.com/106774411/225998098-edea64eb-d401-400c-827c-55c9bfc2907d.jpg)

These are the 4 screens you go through when you turn backup on

**Reply by alienatedsec** [reply]
What are your firmware version @mart-mac and the inverter model?

**Reply by mart-mac** [reply]
Its a RHI-6K-48ES-5G, it's about 10 months old
Software version 3C0032

**Reply by alienatedsec** [reply]
@mart-mac Makes sense - The same model, and I am on `A10032` and its called `BackupReserve`

![image](https://user-images.githubusercontent.com/73167064/226022683-31c9a49d-27b6-4232-bd4c-4949475ad0e4.png)

**Reply by mart-mac** [reply]
Good old solis changing the name, lol.  Have you found all the registry info now

**Reply by alienatedsec** [reply]
submitted the PR #357

**Reply by wills106** [reply]
I have just released[2023.03.2b16](https://github.com/wills106/homeassistant-solax-modbus/releases/tag/2023.03.2b16) with the missing "Backup Mode SOC" & "Battery Minimum SOC"
